src/Simulation/Native/prerequisites.ps1

```diff
@@ -2,15 +2,29 @@
 # Licensed under the MIT License.
 
 if (($IsMacOS) -or ((Test-Path Env:AGENT_OS) -and ($Env:AGENT_OS.StartsWith("Darwin")))) {
-    # building with gcc-9 succeeds but some of the unit tests fail
-    Write-Host "Install gcc-7 as pre-req for building native simulator on MacOS"
-    # temporary workaround for Bintray sunset
-    # remove this after Homebrew is updated to 3.1.1 on MacOS image, see:
-    # https://github.com/actions/virtual-environments/blob/main/images/macos/macos-10.15-Readme.md
-    brew update
-    brew install gcc@7
-} else {
-    Write-Host "No pre-reqs for building native simulator on platforms other than MacOS"
+    brew install libomp
+} elseif (($IsWindows) -or ((Test-Path Env:/AGENT_OS) -and ($Env:AGENT_OS.StartsWith("Win")))) {
+    if (!(Get-Command clang        -ErrorAction SilentlyContinue) -or `
+        (Test-Path Env:/AGENT_OS)) {
+        choco install llvm --version=11.1.0 --allow-downgrade
+        Write-Host "##vso[task.setvariable variable=PATH;]$($env:SystemDrive)\Program Files\LLVM\bin;$Env:PATH"
+    }
+    if (!(Get-Command ninja -ErrorAction SilentlyContinue)) {
+        choco install ninja
+    }
+    if (!(Get-Command cmake -ErrorAction SilentlyContinue)) {
+        choco install cmake
+    }
+    refreshenv
+}
+else {
+    if (Get-Command sudo -ErrorAction SilentlyContinue) {
+        sudo apt update
+        sudo apt-get install -y clang-11
```

Comment:
I think this should include `ninja-build` for completeness (like https://github.com/microsoft/qsharp-runtime/blob/main/src/Qir/Runtime/prerequisites.ps1#L33)

Reply:
same below in the other branch.

Reply:
I noticed that on Linux and Mac the CMake uses Clang (either by default on Mac or the one specified in the command line args on Linux), without Ninja.
But on Windows, without Ninja the CMake continues to use MSVC even if the compiler is specified as a command line arg. The only way I found to switch the CMake on Windows from MSVC to Clang is by using the Ninja generator (if you know a better way then feel free to share). That is why _for native simulator_ Ninja is only used on Win.
I believe it also makes sense to exclude Ninja _for QIR RT and its tests_ on Mac and Linux. I opened #882 for that.

Reply:
I'll comment on the issue so we can discuss more there, but I don't think I agree. I prefer to use the same generator across all platforms so we can have a more uniform configuration experience. I'll elaborate on the issue itself and then once we determine what course we'd prefer to take either remove Ninja for the QIR Runtime or add it here. Either way, no changes for this PR are required.